var/spack/repos/builtin/packages/spdk/package.py

```diff
@@ -0,0 +1,112 @@
+    variant("pmdk", default=False, description="Build persistent memory bdev")
+    variant("rbd", default=False, description="Build Ceph RBD bdev module")
+    variant(
+        "rdma", default=False, description="Build RDMA transport for NVMf target and initiator"
+    )
+    variant("shared", default=False, description="Build spdk shared libraries")
+    variant("uring", default=False, description="Build I/O uring bdev")
+    variant(
+        "virtio", default=False, description="Build vhost initiator and virtio-pci bdev modules"
+    )
+    variant("vhost", default=False, description="Build vhost target")
+    variant("vtune", default=False, description="Profile I/O under Intel VTune Amplifier XE")
+
+    mods = (
+        "crypto",
+        "dpdk",
+        "iscsi-initiator",
+        "ocf",
+        "pmdk",
+        "rbd",
+        "rdma",
+        "shared",
+        "uring",
+        "vhost",
+        "virtio",
+        "vtune",
+    )
+
+    depends_on("dpdk@22.11:", when="+dpdk")
+    depends_on("fio@3.33", when="+fio")
+    depends_on("libaio")
+    depends_on("meson")
+    depends_on("nasm@2.12.02:", type="build")
+    depends_on("numactl")
+    depends_on("py-pyelftools")
+    depends_on("rdma-core", when="+rdma")
+
+    def configure_args(self):
+        spec = self.spec
+        config_args = ["--disable-tests", "--disable-unit-tests", "--disable-apps"]
+
+        if "+fio" in spec:
+            config_args.append("--with-fio={0}".format(spec["fio"].prefix))
+
+        for mod in self.mods:
+            if "+" + mod in spec:
+                config_args.append("--with-{0}".format(mod))
+            else:
+                config_args.append("--without-{0}".format(mod))
+
+        return config_args
+
+    def install_dpdk_files(self):
+        spec = self.spec
+        prefix = self.prefix
+
+        dpdk_build_dir = join_path(spec["dpdk"].prefix, "lib")
+
+        install_tree(join_path(dpdk_build_dir, "pkgconfig"), join_path(prefix.lib, "pkgconfig"))
+
+        for file in os.listdir(dpdk_build_dir):
+            f = join_path(dpdk_build_dir, file)
+            if os.path.isfile(f):
+                install(f, prefix.lib)
+
+        mkdir(join_path(prefix.include, "dpdk"))
+        install_tree(join_path(spec["dpdk"].prefix, "include"), join_path(prefix.include, "dpdk"))
+
+    @run_after("install")
+    def install_additional_files(self):
+        spec = self.spec
+        prefix = self.prefix
+        if "+dpdk" in spec:
+            self.install_dpdk_files()
+        # Copy the config.h file, as some packages might require it.
+        mkdir(prefix.share)
+        mkdir(join_path(prefix.share, "spdk"))
+        install_tree("examples/nvme/fio_plugin", join_path(prefix.share, "spdk", "fio_plugin"))
+        install_tree("include", join_path(prefix.share, "spdk", "include"))
+        install_tree("scripts", join_path(prefix.share, "spdk", "scripts"))
```

Comment:
I think we can remove all these if we do the dpdk dependency method, this way there won't be any conflict and that should allow you to build. You'll just need to track the `dpdk` version used whenever you add a newer version of `spdk`.
```suggestion

```